### ISSUE TITLE
Feature(s) Request: HTML Input Fields

### ISSUE THREAD

**Issue by Smolations** [swagger-api/swagger-ui#1356]
I'd like to make a feature request regarding input elements in the UI.

The current implementation shows input fields when parameters are all form data, as well as when path, header, and/or query params are present, depending on the param's type. My request is actually a collection of smaller requests:

1. Define a `border-radius` for inputs and textareas. When selects are present, this will give the UI a more consistent look.
2. Instead of using the placeholder attribute to indicate required fields (setting the value to "(required)"), use an asterisk or some other marker _in the label_. This will allow for an example (or default if specified) to show up as the placeholder. This change both preserves the indication that a field is required and allows for users to see the required type of data to be used without having to scan to the right to see the data type under the "Parameter Type" column.
3. Already mentioned in the first request, use appropriate, more informative values for placeholders after changing the location (and possibly type) of indicator for required fields.
4. Use HTML5 inputs for appropriate types. This could include smart assumptions to render a field of type="email" (to which you could also add a pattern attribute if the pattern property is used in that param's schema) or direct translations (e.g. `<input type="number" min="{value of schema's min property if present}" ... />`). 

The browser supplies all sorts of helpful functionality when using HTML web components. I think it will help give swagger an even more modern feel while making it easier for users to input data for requests. As an example, if a parameter is supposed to be an integer, it's attributes can be tweaked to take values from the schema (if provided) like this: `<input type="number" min="1" max="5"/>`

![](http://g.recordit.co/2tJMMYNsv6.gif)

Thank you for your consideration!  :smile:

**p.s.** If my time opens up in the next several days and you guys think this is a good idea, I would even be willing to submit a PR implementing these changes (with any suggestions/direction from the swagger-ui team).

**p.p.s.** Also, whether I work on it or not, if an initial exploration of these changes gets the green light, I am willing to put a list together of all of the input types/attributes which could feasibly be used by incoming data and post them in this thread.

**Comment by webron** [comment]
@Smolations - that's a really nice set of ideas. My only potential concern is browser compatibility, as I'm not sure which versions of IE specifically would support it and what would happen for those who don't.

**Comment by Smolations** [comment]
@webron What if [an html5 shim](https://github.com/aFarkas/html5shiv) is used?

**Comment by webron** [comment]
That may be an interesting option considering right now we still support IE8/IE9.

**Comment by Smolations** [comment]
I will do some more research, and maybe some tinkering, and will update this thread when I have more information. :+1:

**Comment by Smolations** [comment]
I think I have found a decent solution. It only requires 2 additional libs and seems to work fine with the current version of jQuery in swagger (only loaded in Chrome so far). The libs are:

[Webshim](https://afarkas.github.io/webshim/demos/) Provides polyfills for many features, most notably form validation features.

[HTML5shiv](https://github.com/aFarkas/html5shiv) This is necessary for Webshim for the following reason (also noted on the Webshim home page):
> IE8- isn't capable of rendering HTML5 elements. Webshims lib in combination with Modernizr or HTML5shiv automatically fixes this issue and adds WAI-ARIA landmarks to enable accessibility of HTML5.

I started to add Modernizr because I thought its features would also be useful, but if we can get things working with just the shiv, Modernizr is probably overkill. I will do some fiddling on a clean branch in my forked repo and see what I can see.
